### PR TITLE
 Mocking module for Iroha Network crate implemented via channels.

### PR DESCRIPTION
**Pull request by humb1t**

### CODE REVIEW

File: iroha_network/src/mock.rs [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+
+pub mod prelude {
+    //! Re-exports important traits and types. Meant to be glob imported when using `iroha_network`.
+
+    #[doc(inline)]
+    pub use crate::{AsyncStream, Network, Request, Response, State};
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (eadventurous) [inline iroha_network/src/mock.rs:258]: Maybe we can can combine the tests in the main and mock modules, and just conditionally compile the import? The tests there duplicate each other.

Comment (humb1t) [reply iroha_network/src/mock.rs:258]: They should duplicate each other to check that both modules work identical. 
Conditional compilation looks like additional build step - I think it's okay to compile both modules during tests, having only one compiled in client app.

If you have an elegant proposal - please describe it in more details.

Comment (eadventurous) [reply iroha_network/src/mock.rs:258]: If we have identical tests in two places, later we may edit one of them and forget about another one. But yes, I don't have a workaround for another build step right now...

Comment (humb1t) [reply iroha_network/src/mock.rs:258]: Moreover - we will need to update some methods code also( When we will work with network more - it should be rewritten a bit

**Pull request merged**